packages/service/src/glint-service.ts

```diff
@@ -92,11 +93,39 @@ export class GlintService implements Service {
   }
 
   getDiagnostics(fileName: string): ts.DiagnosticWithLocation[] {
-    const diagnostics = this.service.getDiagnostics(fileName);
+    const diagnostics = this.service
+      .getDiagnostics(fileName)
+      .map((diagnostic) => this.convertLSPDiagnosticToTs(fileName, diagnostic));
 
-    return diagnostics.map((diagnostic) => {
-      return this.convertLSPDiagnosticToTs(fileName, diagnostic);
-    });
+    return [...diagnostics, ...this.getAdditionalDiagnostics(fileName)];
+  }
+
+  getAdditionalDiagnostics(fileName: string): DiagnosticWithLocation[] {
+    const sourceFile = this.getSourceFile(fileName);
+    const diagnostics: DiagnosticWithLocation[] = [];
+
+    const visitEveryNode = (node: Node): Node | undefined => {
+      if ((isFunctionDeclaration(node) || isMethodDeclaration(node)) && node.name && !node.type) {
+        diagnostics.push({
+          file: sourceFile,
+          start: node.getStart(),
+          length: node.getEnd() - node.getStart(),
+          category: ts.DiagnosticCategory.Suggestion,
+          code: 7050,
+          messageText:
+            (isFunctionDeclaration(node) ? `Function` : `Method`) +
+            ` '${node.name.getText()}' lacks a return-type annotation.`,
+        });
+
+        return undefined;
+      }
+
+      return forEachChild(node, visitEveryNode);
+    };
+
+    visitEveryNode(sourceFile);
+
+    return diagnostics;
```

Comment:
Have you considered making an abstract base class with this method in there, so we can share the implementation?

Reply:
Just want to keep less changes in this PR. Will "merge" those classes next time once we confirm everything working